File: packages/engine/src/cpus/6502/instructions/cld.js
import { CycleType } from '..'

/**
 * Clear Decimal Mode
 *
 * @example
 * MODE          SYNTAX        HEX LEN TIM
 * Implied       CLD           $D8  1   2
 */
export function* cld() {
  this.decimalMode = false
  yield CycleType.GET
}

File: packages/engine/src/cores/famibob/cpu/instructions/cld.zig
const std = @import("std");
const CPU = @import("../cpu.zig").CPU;
const Addressing = @import("../opcode.zig").Addressing;

// Clear Decimal Mode
//
// MODE          SYNTAX        HEX LEN TIM
// Implied       CLD           $D8  1   2

pub fn cld(self: *CPU, addressing: Addressing) void {
    _ = addressing;
    self.d = false;
}
